src/jlapi.c

```diff
@@ -492,6 +492,9 @@ static int exec_program(char *program)
         jl_load(jl_main_module, program);
     }
     JL_CATCH {
+        // TODO: It is possible for this output
+        //       to be mangled due to `jlbacktrace`
+        //       printing directly to STDERR_FILENO.
```

Comment:
Is it possible to resolve this as you've done elsewhere then?

Reply:
It wasn't immediately clear to me how to do so. The code mixes `JL_STDERR` and `jl_stderr_objs` and I presume it does so for a reason. I am not sure if it is gurantueed that hose two streams will be the same.

Some git archeology shows that this is code from the ancient times.

Reply:
Oh yes I see. So it seems that `errs` could certainly be pointing to some other stream if `redirect_stderr` was called.  I think the fix here, in principle, is to have some shim to call `display_error()` rather than printing the current exception and backtrace separately.  If `display_error` doesn't exist or fails, then the fallback path of `shown_err == 0` would be taken.